inference-engine/tests/functional/inference_engine/keep_constant_inputs_tests.cpp

```diff
@@ -0,0 +1,156 @@
+            if (auto dtsOp = std::dynamic_pointer_cast<const ::ngraph::opset3::DepthToSpace>(node)) {
+                return dtsOp->input_value(0).get_shape().size() <= 5lu &&
+                       dtsOp->input_value(0).get_shape().size() == dtsOp->get_output_shape(0).size();
+            }
+
+            // SpaceToDepth node implementation supports only equal input/output tensors with rank <= 5
+            if (auto stdOp = std::dynamic_pointer_cast<const ::ngraph::opset3::SpaceToDepth>(node)) {
+                return stdOp->input_value(0).get_shape().size() <= 5lu &&
+                       stdOp->input_value(0).get_shape().size() == stdOp->get_output_shape(0).size();
+            }
+
+            return std::dynamic_pointer_cast<const ::ngraph::opset2::Gelu>(node) ||
+                   std::dynamic_pointer_cast<const ::ngraph::opset2::BatchToSpace>(node) ||
+                   std::dynamic_pointer_cast<const ::ngraph::opset2::SpaceToBatch>(node) ||
+                   std::dynamic_pointer_cast<const ::ngraph::opset3::ShuffleChannels>(node);
+        };
+        auto nGraphFunc = clonedNetwork->getFunction();
+        // Disable shape inference (WA for generic operations)
+        ::ngraph::op::GenericIE::DisableReshape noReshape(nGraphFunc);
+
+        // Note: instead of running all Conversion Transformations you can make up your own transformation pipeline
+        ngraph::pass::CommonOptimizations(transformations_callback).run_on_function(nGraphFunc);
+        ngraph::pass::ConvertOpSet3ToOpSet2(transformations_callback).run_on_function(nGraphFunc);
+        ngraph::pass::ConvertOpSet2ToOpSet1(transformations_callback).run_on_function(nGraphFunc);
+        ngraph::pass::ConvertOpSet1ToLegacy(transformations_callback).run_on_function(nGraphFunc);
+        clonedNetwork = InferenceEngine::details::convertFunctionToICNNNetwork(nGraphFunc, *clonedNetwork, keep_constant_inputs);
+    }
+}
+
+TEST(KeepConstantInputsTests, ConvertConvolutionPoolReluNetworkWithTrue) {
+    std::shared_ptr <ngraph::Function> f_ptr;
+    f_ptr = ngraph::builder::subgraph::makeConvPoolRelu();
+    InferenceEngine::CNNNetwork originalNetwork(f_ptr);
+    std::shared_ptr<InferenceEngine::ICNNNetwork> clonedNetwork = InferenceEngine::cloneNetwork(originalNetwork);
+    transformNetwork(clonedNetwork, true);
+    InferenceEngine::CNNNetwork convertedNetwork(clonedNetwork);
+    std::cout << "Check for conversion of ConvolutionPoolRelu Network with keep_constant_inputs = true" << std::endl;
+    ASSERT_GT(numberOfInputsForLayerInCNNNetwork(convertedNetwork, "Convolution"), 1);
+}
+
+TEST(KeepConstantInputsTests, ConvertConvolutionPoolReluNetworkWithFalse) {
+    std::shared_ptr <ngraph::Function> f_ptr;
+    f_ptr = ngraph::builder::subgraph::makeConvPoolRelu();
+    InferenceEngine::CNNNetwork originalNetwork(f_ptr);
+    std::shared_ptr<InferenceEngine::ICNNNetwork> clonedNetwork = InferenceEngine::cloneNetwork(originalNetwork);
```

Comment:
You don't need to create a copy of nGraph function. Please remove (in all tests).

Reply:
Done.